### PR TITLE
[Merged by Bors] - feat (Data/Polynomial/RingDivision): add lemmas for subtraction and negation

### PR DESCRIPTION
**Pull request by goens**
Adds two lemmas that show that negation and subtraction commute with polynomial division.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/RingDivision.lean [hunk 1]
```diff
@@ -105,6 +105,19 @@ def modByMonicHom (q : R[X]) : R[X] →ₗ[R] R[X] where
...
+    have H' : - (p %ₘ mod) = (-1 : R) • (p %ₘ mod) := by norm_num
+    rw[H, H']
+    apply smul_modByMonic (R := R) (c := -1) (p := p) (q := mod)
+
+@[simp]
+theorem sub_modByMonic (a b mod : R[X]) : (a - b) %ₘ mod = a %ₘ mod - b %ₘ mod := by
+  ring_nf
+  repeat rw[sub_eq_add_neg]
+  simp[Polynomial.add_modByMonic]
+  rw[neg_modByMonic]
```
Comment (ChrisHughes24) [inline Mathlib/Data/Polynomial/RingDivision.lean:119]:
```suggestion
  rw [sub_eq_add_neg, add_modByMonic, neg_modByMonic, sub_eq_add_neg]
```

File: Mathlib/Data/Polynomial/RingDivision.lean [hunk 2]
```diff
@@ -105,6 +105,19 @@ def modByMonicHom (q : R[X]) : R[X] →ₗ[R] R[X] where
   map_smul' := smul_modByMonic
 #align polynomial.mod_by_monic_hom Polynomial.modByMonicHom
 
+theorem neg_modByMonic (p mod : R[X]) : (-p) %ₘ mod = - (p %ₘ mod) := by
+    have H : -p = (-1 : R) • p := by norm_num
+    have H' : - (p %ₘ mod) = (-1 : R) • (p %ₘ mod) := by norm_num
+    rw[H, H']
+    apply smul_modByMonic (R := R) (c := -1) (p := p) (q := mod)
```
Comment (ChrisHughes24) [inline Mathlib/Data/Polynomial/RingDivision.lean:112]:
```suggestion
  have H : -p = (-1 : R) • p := by simp
  have H' : - (p %ₘ mod) = (-1 : R) • (p %ₘ mod) := by simp
  rw [H, H', smul_modByMonic]
```

File: Mathlib/Data/Polynomial/RingDivision.lean [hunk 3]
```diff
@@ -105,6 +105,19 @@ def modByMonicHom (q : R[X]) : R[X] →ₗ[R] R[X] where
...
+
+@[simp]
```
Comment (ChrisHughes24) [inline Mathlib/Data/Polynomial/RingDivision.lean:114]:
```suggestion
```

**Review by ChrisHughes24 — COMMENTED** [review]
Also, both lemmas should probably be moved so that they are immediately after `smul_modByMonic`

File: Mathlib/Data/Polynomial/RingDivision.lean (hunk 2 above)
Comment (ChrisHughes24) [inline Mathlib/Data/Polynomial/RingDivision.lean:112]: Actually this is better
```suggestion
theorem neg_modByMonic (p mod : R[X]) : (-p) %ₘ mod = - (p %ₘ mod) :=
  (modByMonicHom mod).map_neg p
```

File: Mathlib/Data/Polynomial/RingDivision.lean (hunk 1 above)
Comment (ChrisHughes24) [inline Mathlib/Data/Polynomial/RingDivision.lean:119]:
```suggestion
theorem sub_modByMonic (a b mod : R[X]) : (a - b) %ₘ mod = a %ₘ mod - b %ₘ mod :=
  (modByMonicHom mod).map_sub a b
```
With these new suggestions, you obviously have to keep the lemmas where they are already.

**Review by ChrisHughes24 — COMMENTED** [review]
I had some better ideas

**Comment by goens** [comment]
thanks!

**Comment by alexjbest** [comment]
maintainer merge

**Review by Vierkantor — APPROVED** [review]
I was wondering if the lemmas should be marked `@[simp]` but apparently neither are the `add` and `smul` versions. So, looks good to me!

bors r+

**Pull request closed without merging**